apis/field_error.go

```diff
@@ -312,20 +312,29 @@ func ErrDisallowedUpdateDeprecatedFields(fieldPaths ...string) *FieldError {
 }
 
 // ErrInvalidArrayValue constructs a FieldError for a repetetive `field`
-// at `index` that has received an invalid string value.
+// at `index` that has received an invalid value.
 func ErrInvalidArrayValue(value interface{}, field string, index int) *FieldError {
 	return ErrInvalidValue(value, CurrentField).ViaFieldIndex(field, index)
 }
 
 // ErrInvalidValue constructs a FieldError for a field that has received an
-// invalid string value.
+// invalid value.
 func ErrInvalidValue(value interface{}, fieldPath string) *FieldError {
 	return &FieldError{
 		Message: fmt.Sprintf("invalid value: %v", value),
 		Paths:   []string{fieldPath},
 	}
 }
 
+// ErrInvalidCombination constructs a FieldError for a field that has received an
+// invalid combination of values.
+func ErrInvalidCombination(bad, value interface{}, key, fieldPath string) *FieldError {
```

Comment:
The ordering of fields seems confusing here and easy to get wrong. 

```suggestion
func ErrInvalidCombination(invalidKey, conflictingKey, conflictingValue interface{}, fieldPath string) *FieldError {
```